calus.js

```diff
@@ -1,228 +1,185 @@
-import Vue from 'vue'
-import { DateTime } from 'luxon'
-
-let defaultTemplate = `
-<div class="month-container" v-bind:class="{ 'month-container--column': displayInColumn }">
-    <div class="month" v-for="month in months" v-bind:data-month="month.time.toFormat('MM/y')">
-        <div class="month__header">
-        <button type="button" class="month__control month__control--prev" v-if="!displayInColumn" v-on:click="scrollMonth(-1)">
-            ‹
-        </button>
-        <div class="month__text">
-            {{ month.time.toFormat(month.isInCurrentYear ? 'MMMM' : 'MMMM y') }}
-        </div>
-        <button type="button" class="month__control month__control--prev" v-if="!displayInColumn" v-on:click="scrollMonth(1)">
-            ›
-        </button>
-        </div>
-        <div class="month__days">
-        <div class="day"
-            v-for="day in month.days"
-            v-bind:data-date="+day.time"
-            v-bind:class="{
-            'is-today': day.isToday,
-            'is-not-today': !day.isToday,
-            'is-available': day.isAvailable,
-            'is-not-available': !day.isAvailable,
-            'is-past': day.isPast,
-            'is-future': day.isFuture,
-            'is-selected': day.isSelected,
-            'is-this-month': day.isThisMonth,
-            'is-different-month': !day.isThisMonth,
-            }"
-            v-on:click="select(day)"
-        >
-            <div class="day__inner">
-                {{ day.time.day }}
-            </div>
-        </div>
-        </div>
-    </div>
-</div>
-`
-export default function calus(options) {
-    let el = options.el || '#calendar'
-    let docEl = el instanceof HTMLElement ? el : document.querySelector(el)
-
-    if (!docEl) throw 'no element found: ' + docEl
-
-    if (options.useDefaultTemplate) {
-        docEl.innerHTML = defaultTemplate
-    }
-
-    return new Vue({
-        el: el,
-        data: {
+import { DateTime, Settings } from 'luxon'
+
+export default {
+    props: ['timeZone', 'availableDates', 'displayInColumn', 'linearDates', 'weekStartsOnSunday', 'onSelect', 'onChangeMonth', 'allowPreviousScroll'],
```

Comment:
Instead of `allowPreviousScroll` change to `allowScrollingOutsideOfDateRange` where by will limit you to the months in `availableDates`

Reply:
Better way of handling the `default` values for the properties https://v2.vuejs.org/v2/guide/components-props.html